Урок 2.

Robilko opened the pull request:
Сверстал два fragment'a. Первый содержит два EditText:
        1) editText логина
        2)editText пароля
        3) кнопка "продолжить" (Login)
![Screenshot_20211213_152001](https://user-images.githubusercontent.com/75025794/145811292-298dc470-6e64-4dc5-8470-b9610b88c9c2.png)

        По нажатию на кнопку "продолжить" логин и пароль пересылаются в LoginPresenter 1 фрагмента (LoginFragment).
        LoginPresenter валидирует введенные данные, что они не пустые и с помощью чичероне кидает на второй экран (UserFragment).
![Screenshot_20211213_152011](https://user-images.githubusercontent.com/75025794/145811365-d708fbd5-c831-4d07-884c-d7bf8f1c0146.png)
![Screenshot_20211213_152022](https://user-images.githubusercontent.com/75025794/145811368-8d628a15-1675-41e9-9e9c-72d146b8b716.png)

        На втором экране прокинул и отобразил в textView логин, который мы ввели editText на предыдущем экране.

![Screenshot_20211213_152035](https://user-images.githubusercontent.com/75025794/145811386-517ca19a-881b-44a6-a583-7039a75260ca.png)

Robilko committed:
Урок 2.
Сверстал два fragment'a. Первый содержит два EditText:
        1) editText логина
        2)editText пароля
        3) кнопка "продолжить" (Login)

        По нажатию на кнопку "продолжить" логин и пароль пересылаются в LoginPresenter 1 фрагмента (LoginFragment).
        LoginPresenter валидирует введенные данные, что они не пустые и с помощью чичероне кидает на второй экран (UserFragment).

        На втором экране прокинул и отобразил в textView логин, который мы ввели editText на предыдущем экране.

Lime94 commented on app/src/main/java/com/example/poplibraries/mvpuser/UserFragment.kt:
```diff
@@ -0,0 +1,38 @@
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        presenter.init()
+        retainInstance = true //временное решение, чтобы пережить переворот экрана в этом фрагменте
```
а зачем тебе переживать, как раз для этого и создан Presenter, чтобы переживать поворот экрана

Robilko replied:
с божьей помощью, сделал иначе

Lime94 commented on app/src/main/java/com/example/poplibraries/mvpuser/UserPresenter.kt:
```diff
@@ -0,0 +1,16 @@
+        viewState.setUserLogin(user.login)
+    }
+
+    fun init() {}
```
init пустой, можно убрать?

Robilko replied:
исправил

Robilko committed:
Урок 2.
Исправил по комментариям преподавателя. +сделал сохранение второго фрагмента через аргументы
